apps/common/behaviours/casework-submission.js

```diff
@@ -1,16 +1,19 @@
 'use strict';
 
+const AuthToken = require('../models/auth-token');
 const CaseworkModel = require('../models/i-casework');
 const StatsD = require('hot-shots');
 const client = new StatsD();
 
 const Compose = func => superclass => class extends superclass {
   prepare() {
     if (typeof func === 'function') {
-      return Object.assign(super.prepare(), func(this.toJSON()));
+      const model = new AuthToken();
+      return model.auth().then(token => {
```

Comment:
Generate auth token here and pass to i-casework model prepare step and payload builders that setup the object that is sent to icasework